source/apiVolontaria/volunteer/permissions.py

```diff
@@ -1,21 +1,70 @@
 from rest_framework import permissions
 
 
-class IsOwnerOrReadOnly(permissions.BasePermission):
+class EventsPermissions(permissions.BasePermission):
     """
     Custom permission to only allow owners of an object to edit it.
-
-    This is used for Participation only for now.
     """
 
+    @staticmethod
+    def if_can_do_actions(request, obj):
+        if_django_perm = False
+
+        if request.method == 'PATCH':
+            if_django_perm = request.user.has_perm(
+                'volunteer.change_event')
+
+        elif request.method == 'DELETE':
+            if_django_perm = request.user.has_perm(
+                'volunteer.delete_event')
+
+        is_manager = request.user in obj.cell.managers.all()
+
+        return if_django_perm or is_manager
+
     def has_object_permission(self, request, view, obj):
         # Read permissions are allowed to any request,
         # so we'll always allow GET, HEAD or OPTIONS requests.
         if request.method in permissions.SAFE_METHODS:
             return True
 
-        # Write permissions are only allowed to the owner of the snippet.
-        if obj.user == request.user or request.user.is_superuser:
+        # Write permissions are only allowed to the owner of the snippet
+        # or to a Cell Manager.
+        can_edit = self.if_can_do_actions(request, obj)
+
+        return can_edit
+
+
+class ParticipationsPermissions(permissions.BasePermission):
+    """
+    Custom permission to only allow owners of an object to edit it.
+    """
+
+    @staticmethod
+    def if_can_do_actions(request, obj):
+        if_django_perm = False
+
+        if request.method == 'PATCH':
+            if_django_perm = request.user.has_perm(
+                'volunteer.change_participation')
+
+        elif request.method == 'DELETE':
+            if_django_perm = request.user.has_perm(
+                'volunteer.delete_participation')
+
+        is_owner = obj.user == request.user
+        is_manager = request.user in obj.event.cell.managers.all()
```

Comment:
Do we want manager to be able to delete participations ? 
I don't see a use case when this can be used.

Reply:
Ok removed